src/DataManagement/MonthlyReport/MonthlyReportCalculator.cs

```diff
@@ -104,22 +104,34 @@ private void CalculateGoalReport()
             }
 
             Result.HasGoals = true;
-            var min = decimal.MaxValue;
-            var minGoal = new Goal();
-            var maxGoal = new Goal();
-            var max = 0M;
-            foreach (var item in Goals)
+           
+          
+
+            var maxGoal = Goals.Aggregate((expensive, next) => next.Amount >= expensive.Amount ? next : expensive);
+            var minGoal = Goals.Aggregate((cheapest, next) => (next.Amount <= cheapest.Amount && next.Amount >= AllSavings)  ? next : cheapest);
+
+            if (maxGoal.Amount < AllSavings && minGoal.Amount < AllSavings)
             {
-                if (item.Amount >= max)
-                {
-                    max = item.Amount;
-                    maxGoal = (Goal)item;
-                }
-
-                if (item.Amount > min || item.Amount <= AllSavings) continue;
-                min = item.Amount;
-                minGoal = (Goal)item;
+                Result.HasGoals = false;
+                return;
             }
```

Comment:
So with aggregate you were able to shorten this much code? Seems like a neat method to use

Reply:
Basically yes, since it allows to avoid ifs and no need to have middle vars such as max and min (but it could be avoided if goals were comparable by IComparable interface by amounts (mayybe we have it but eh :D)).